src/views/ConfigEditor.tsx

```diff
@@ -14,13 +19,30 @@ const ConfigEditor = (props: ConfigEditorProps) => {
   const { jsonData, secureJsonData, secureJsonFields } = options;
   const secureSettings = (secureJsonData || {}) as GitHubSecureJsonData;
   const styles = useStyles2(getStyles);
+  const WIDTH_LONG = 40;
+  const authOptions = [
+    { label: 'Personal Access Token', value: 'personal-access-token' },
+    { label: 'GitHub App', value: 'github-app' },
+  ];
+  const licenseOptions = [
+    { label: 'Basic', value: 'github-basic' },
+    { label: 'Enterprise', value: 'github-enterprise' },
+  ];
 
-  const [selectedLicense, setSelectedLicense] = useState(jsonData.githubUrl ? 'github-enterprise' : 'github-basic');
   const [isOpen, setIsOpen] = useState(true);
+  const [selectedLicense, setSelectedLicense] = useState(jsonData.githubUrl ? 'github-enterprise' : 'github-basic');
 
-  const onSettingUpdate =
-    (prop: string, set = true) =>
-    (event: ChangeEvent<HTMLInputElement>) => {
+  useEffect(() => {
+    // set the default auth type if its a new datasource and nothing is set
+    if (!jsonData.selectedAuthType) {
+      onAuthChange('personal-access-token');
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I think this would correctly work if you added `[jsonData.selectedAuthType]` and we wouldn't have to use `// eslint-disable-next-line react-hooks/exhaustive-deps`